### PR TITLE
Update to Pytest 6.0

### PR DESCRIPTION
**Pull request by ashb**
And pytest 6 removed a class that the rerunfailures plugin was using, so
we have to upgrade that too.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]

**Comment by ashb** [comment]
Lets see if full suite passes here -- I tested a few things locally and didn't notice any problems.

File: setup.py [hunk 1]
```diff
@@ -502,10 +502,10 @@ def get_sphinx_theme_version() -> str:
     'pre-commit',
     'pylint==2.6.0',
     'pysftp',
-    'pytest',
+    'pytest~=6.0',
```
Comment (kaxil) [inline setup.py:505]: We are already using the latest pytest:

https://github.com/apache/airflow/blob/ae2e0275bde2e551a63e2ee9b1c56538535e7657/constraints-3.6.txt#L327

https://github.com/apache/airflow/blob/ae2e0275bde2e551a63e2ee9b1c56538535e7657/constraints-3.7.txt#L322

**Comment by ashb** [comment]
Constraints file is already using this, so should be fine to merge.

File: setup.py (hunk 1 above)
Comment (ashb) [reply setup.py:505]: In constraints yes, this enforces it for anyone now using the constraint files too.

**Pull request merged**